packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -20,6 +33,40 @@ def _ensure_routable():
     return _ensure_routable
 
 
+def test_if_overlay_ok(cluster):
+    def _check_overlay(hostname, port):
+        uri = 'http://{}:{}/overlay-agent/overlay'.format(hostname, port)
+        resp = requests.get(uri).json()
+        overlays = resp['overlays']
+        assert(len(overlays) > 0)
+        for overlay in overlays:
+            assert(overlay['state']['status'] == 'STATUS_OK')
+        return True
+
+    @retrying.retry(wait_fixed=2000,
```

Comment:
just move this retry decorator on top of the test and dont use this inner function at all